File: java/cours/src/session2/td/heritageMultiple/NatParIntDecimal.java
package session2.td.heritageMultiple;

import session1.td.Nat;

public class NatParIntDecimal extends IntPositif implements AlgebreNatDecimal {

	public NatParIntDecimal(int rep) {
		super(rep);
	}
	
	@Override
	public Nat creerNatAvecValeur(int val) {
		return new NatParIntDecimal(val);
	}
	
	@Override
	public String toString() {
		return Integer.toString(this.val());
	}
	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof Nat))
			return false;
		Nat x = (Nat)obj;
		return this.val() == x.val();
	}
}

File: typescript/session2/heritageMultiple/NatParIntDecimal.ts
import {IntPositif} from "./IntPositif";
import {AlgebreNatDecimal} from "./AlgebreNatDecimal";
import {Nat} from "../../session1/td/Nat/Nat";

export class NatParIntDecimal extends IntPositif, AlgebreNatDecimal {

    constructor(rep : number) {
        super(rep);
    }

    creerNatValue( val : number) : Nat{
        return new NatParIntDecimal(val);
    }

    toString() : string{
        return '' + this.getVal();
    }

    equals(obj : Object) : boolean{
        if(!(obj instanceof Nat))
            return false;
        let x = <Nat>obj;
        return this.getVal() == x.getVal();
    }
}
